src/Type/Php/MbConvertEncodingFunctionReturnTypeExtension.php

```diff
@@ -25,18 +24,16 @@ public function getTypeFromFunctionCall(
 		Scope $scope,
 	): ?Type
 	{
-		if (!isset($functionCall->getArgs()[0])) {
+		$args = $functionCall->getArgs();
+		if (!isset($args[0])) {
 			return null;
 		}
 
-		$argType = $scope->getType($functionCall->getArgs()[0]->value);
+		$argType = $scope->getType($args[0]->value);
 		$isString = $argType->isString();
 		$isArray = $argType->isArray();
-		$compare = $isString->compareTo($isArray);
-		if ($compare === $isString) {
-			return new StringType();
-		} elseif ($compare === $isArray) {
-			return new ArrayType(new IntegerType(), new StringType());
+		if ($isString->yes() || $isArray->yes()) {
+			return new UnionType([$argType, new ConstantBooleanType(false)]);
```

Comment:
In all versions, the `mb_convert_encoding()` function throws an error for invalid encodings, so it makes sense to use BenevolentUnionType instead of Union. https://3v4l.org/YuXmF

```suggestion
			return new BenevolentUnionType([$argType, new ConstantBooleanType(false)]);
```

Using a benevolent union, we can pass it to function calls expecting strings without any extra assertions, like this: https://phpstan.org/r/fbcfe268-adc6-48cd-9f35-c66b3117c269

Reply:
I am not sure about this change because I don't know if there are other ways this function can fail.

Reply:
[As the PHP manual says](https://www.php.net/manual/en/function.mb-convert-encoding.php#refsect1-function.mb-convert-encoding-errors), the function will only raise an `E_WARNING` if an invalid encoding is specified.

> ## Errors/Exceptions
> As of PHP 8.0.0, a [ValueError](https://www.php.net/manual/en/class.valueerror.php) is thrown if the value of to_encoding or from_encoding is an invalid encoding. Prior to PHP 8.0.0, a [E_WARNING](https://www.php.net/manual/en/errorfunc.constants.php#constant.e-warning) was emitted instead.

This function behaves this way in all versions, so it will never return `null`, even if an invalid byte sequence is input.

 * https://github.com/php/php-src/blob/PHP-7.3/ext/mbstring/mbstring.c#L3057-L3060
 * https://github.com/php/php-src/blob/PHP-8.4/ext/mbstring/mbstring.c#L2728-L2729

Reply:
There is still use cases which returns false, like
```
mb_convert_encoding("\x80\x81\x82", "UTF-8", 'auto')
```

So someone calling `mb_convert_encoding($string, $from, $to)` should be still advised to use a false check.

UnionType might be preferred then.

Reply:
reverted back to regular union type